web/src/beta/features/Editor/Map/DataSourceLayerCreator/ThreeDTiles/index.tsx

```diff
@@ -3,29 +3,120 @@ import {
   const t = useT();
+
   const [value, setValue] = useState("");
+  const [sourceType, setSourceType] = useState<SourceType>("osm-buildings");
+  const [googleMapsApiKey, setGoogleMapsApiKey] = useState("");
+  const googlePhotorealistic = sourceType === "google-photorealistic";
+
+  const renderGooglePhotorealisticInput = useMemo(() => {
+    if (googlePhotorealistic) {
+      return (
+        <InputGroup
+          label={
+            <>
+              {t("Google Maps API Key ")} ( {t("You can apply a key ")}
+              <LinkWrapper
+                to="https://developers.google.com/maps/documentation/javascript/get-api-key"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {t("here")}
+              </LinkWrapper>
+              )
+            </>
+          }
+        >
+          <InputsWrapper>
+            <TextInput
+              value={googleMapsApiKey}
+              onChange={(value) => setGoogleMapsApiKey(value)}
+            />
+          </InputsWrapper>
+        </InputGroup>
+      );
+    } else return undefined;
+  }, [googleMapsApiKey, googlePhotorealistic, t]);
+
+  const renderUrlInput = useMemo(() => {
+    if (sourceType === "url") {
+      return (
+        <InputGroup label={t("Resource URL")}>
+          <InputsWrapper>
+            <TextInput
+              placeholder="https://"
+              value={value}
+              onChange={(value) => setValue(value)}
+            />
+          </InputsWrapper>
+        </InputGroup>
+      );
+    } else return undefined;
+  }, [sourceType, t, value]);
+
+  const dataSourceOptions: DataSourceOptType = useMemo(
+    () => [
+      { label: t("Cesium OSM 3D Tiles"), value: "osm-buildings" },
+      {
+        label: t("Google Photorealistic 3D Tiles"),
+        value: "google-photorealistic",
+        children: renderGooglePhotorealisticInput
+      },
+      {
+        label: t("URL"),
+        value: "url",
+        children: renderUrlInput
+      }
+    ],
+    [renderGooglePhotorealisticInput, renderUrlInput, t]
+  );
+
+  const handleDataSourceTypeChange = useCallback((newValue: string) => {
+    setSourceType(newValue as SourceType);
+    setValue("");
+    setGoogleMapsApiKey("");
+  }, []);
+
+  const title = useMemo(() => {
+    if (googlePhotorealistic) {
+      return t("Google Photorealistic 3D Tiles");
+    } else if (sourceType === "osm-buildings") {
+      return t("Cesium OSM 3D Tiles");
+    } else {
+      return generateTitle(value);
+    }
+  }, [googlePhotorealistic, sourceType, t, value]);
 
   const handleSubmit = () => {
     onSubmit({
       layerType: "simple",
       sceneId,
-      title: generateTitle(value),
+      title,
       visible: true,
       config: {
         data: {
           url: value !== "" ? value : undefined,
-          type: "3dtiles"
+          serviceTokens: {
+            googleMapApiKey: googleMapsApiKey || undefined
+          },
```

Comment:
```suggestion
          ...(googlePhotorealistic
            ? {
                serviceTokens: {
                  googleMapApiKey: googleMapsApiKey || undefined
                }
              }
            : {}),
```
I think better add this only when needed. For most layers we don't need to set this property.